### PR TITLE
Interpreter directive

### PR DESCRIPTION
**Pull request by diokey**
Interpreter directive comment was parsed as both interpreter directive and comment. As a result the formatter formatted it as a comment. 

This PR updates the parser not to parse the interpreter directive as a comment and also adds a builder for interpreter directive

Closes https://github.com/romejs/rome/issues/466

### CODE REVIEW

File: packages/@romejs/js-parser/tokenizer/index.ts [hunk 1]
```diff
@@ -410,6 +410,34 @@ export function skipLineComment(parser: JSParser, startSkip: number): AnyComment
...
+			ch !== charCodes.lineFeed &&
+			ch !== charCodes.carriageReturn &&
+			ch !== charCodes.lineSeparator &&
+			ch !== charCodes.paragraphSeparator &&
+			bumpIndex(parser) < parser.length
+		) {
+			ch = parser.input.charCodeAt(getIndex(parser));
+		}
+	}
+	const loc = parser.finishLocAt(startPos, parser.getPositionFromState());
```
Comment (sebmck) [inline packages/@romejs/js-parser/tokenizer/index.ts:432]: I wonder if there's a way that we can clean this up. Maybe we can pull it out into a method and share it with `skipLineComment`.

Comment (diokey) [reply packages/@romejs/js-parser/tokenizer/index.ts:432]: Yup, I'll look into that

File: packages/@romejs/js-formatter/builders/core/Program.ts [hunk 2]
```diff
@@ -18,6 +18,10 @@ export default function Program(builder: Builder, node: Program): Token {
 		tokens.push(hardline);
 	}
 
+	if (node.interpreter) {
```
Comment (sebmck) [inline packages/@romejs/js-formatter/builders/core/Program.ts:21]: I think we'll also want to add a formatter option to allow the inclusion of interpreter directives. If we're generating a bundle we'll include them which would become a syntax error.

You could add a `allowInterpreterDirective` property to `packages/@romejs/js-formatter/index.ts`, and just make it default to `true` then in `packages/@romejs/js-compiler/api/compile.ts` pass it into `formatJS` as `false`.

Comment (diokey) [reply packages/@romejs/js-formatter/builders/core/Program.ts:21]: ooh right. Will do. Thanks

**Comment by sebmck** [comment]
Awesome! This looks great! Thanks for taking the time to do this.

**Pull request merged**